__tests__/app.test.js

```diff
@@ -236,3 +261,151 @@ describe("GET /api/articles/:article_id/comments", () => {
+			.send({
+				body: "test comment",
+			})
+			.expect(400)
+			.then((data) => {
+				expect(data.body.msg).toBe("Bad request!");
+			});
+	});
+	it("should send the correct status code back (400) when passed a comment object with too many key value pairs", () => {
+		return request(app)
+			.post("/api/articles/4/comments")
+			.send({
+				username: "lurker",
+				body: "test comment",
+				info: "please cause an error :)",
+			})
+			.expect(400)
+			.then((data) => {
+				expect(data.body.msg).toBe("Bad request!");
+			});
+	});
+	it("should send the correct status code back (404) when trying to post a comment to an article that doesnt exist ", () => {
+		return request(app)
+			.post("/api/articles/440/comments")
+			.send({
+				username: "lurker",
+				body: "test comment",
+			})
+			.expect(404)
+			.then((data) => {
+				expect(data.body.msg).toBe("Not found!");
+			});
+	});
+	it("should send the correct status code back (400) when passed a comment object that contains one incorrect keys", () => {
+		return request(app)
+			.post("/api/articles/4/comments")
+			.send({
+				username: "lurker",
+				bdy: "test comment",
+			})
+			.expect(400)
+			.then((data) => {
+				expect(data.body.msg).toBe("Bad request!");
+			});
+	});
+	it("should send the correct status code back (400) when passed a comment object that contains two incorrect keys", () => {
+		return request(app)
+			.post("/api/articles/4/comments")
+			.send({
+				usename: "lurker",
+				bdy: "test comment",
+			})
+			.expect(400)
+			.then((data) => {
+				expect(data.body.msg).toBe("Bad request!");
+			});
+	});
+	it("should send the correct status code back (404) when trying to post a comment to an article id that is of the wrong format - article_id should be an integer ", () => {
+		return request(app)
+			.post("/api/articles/four/comments")
+			.send({
+				username: "lurker",
+				body: "test comment",
+			})
+			.expect(400)
+			.then((data) => {
+				expect(data.body.msg).toBe("Bad request!");
+			});
+	});
+	it("should send the correct status code back (400) when trying to post a comment with a username that doesnt exist in the database", () => {
+		return request(app)
+			.post("/api/articles/4/comments")
+			.send({
+				username: "j4ckth3h4cker",
+				body: "test comment",
+			})
+			.expect(400)
+			.then((data) => {
+				expect(data.body.msg).toBe("User not found!");
+			});
+	});
+	it("should return the comment object back once posted and it should contain all the expected key value pairs", () => {
+		return request(app)
+			.post("/api/articles/4/comments")
+			.send({
+				username: "rogersop",
+				body: "Hello, this is a comment from Roger",
+			})
+			.expect(201)
+			.then((data) => {
+				expect(Object.keys(data.body)).toInclude(
+					"comment_id",
+					"body",
+					"article_id",
+					"author",
+					"votes",
+					"created_at"
+				);
+				expect(data.body).toMatchObject({
+					comment_id: expect.any(Number),
```

Comment:
we can figure out what the comment id value should be by looking at how many comments there currently are in the database (i.e. after seeding)